tests/components/smartthings/test_config_flow.py

```diff
@@ -4,207 +4,266 @@
 from aiohttp import ClientResponseError
 from asynctest import Mock, patch
 from pysmartthings import APIResponseError
+from pysmartthings.installedapp import format_install_url
 
 from homeassistant import data_entry_flow
 from homeassistant.components.smartthings import smartapp
 from homeassistant.components.smartthings.config_flow import SmartThingsFlowHandler
 from homeassistant.components.smartthings.const import (
+    CONF_APP_ID,
     CONF_INSTALLED_APP_ID,
-    CONF_INSTALLED_APPS,
     CONF_LOCATION_ID,
+    CONF_OAUTH_CLIENT_ID,
+    CONF_OAUTH_CLIENT_SECRET,
     CONF_REFRESH_TOKEN,
     DOMAIN,
 )
-from homeassistant.const import HTTP_FORBIDDEN, HTTP_NOT_FOUND
-from homeassistant.setup import async_setup_component
+from homeassistant.const import CONF_ACCESS_TOKEN, HTTP_FORBIDDEN, HTTP_NOT_FOUND
 
 from tests.common import MockConfigEntry, mock_coro
 
 
-async def test_step_user(hass):
-    """Test the access token form is shown for a user initiated flow."""
+async def test_step_import(hass):
+    """Test import returns user."""
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    result = await flow.async_step_user()
+
+    result = await flow.async_step_import()
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == "user"
+    assert result["description_placeholders"][
+        "webhook_url"
+    ] == smartapp.get_webhook_url(hass)
 
 
-async def test_step_init(hass):
-    """Test the access token form is shown for an init flow."""
+async def test_step_user(hass):
+    """Test the webhook confirmation is shown."""
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    result = await flow.async_step_import()
+
+    result = await flow.async_step_user()
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["step_id"] == "user"
+    assert result["description_placeholders"][
+        "webhook_url"
+    ] == smartapp.get_webhook_url(hass)
 
 
-async def test_base_url_not_https(hass):
-    """Test the base_url parameter starts with https://."""
+async def test_step_user_aborts_invalid_webhook(hass):
+    """Test flow aborts if webhook is invalid."""
     hass.config.api.base_url = "http://0.0.0.0"
     flow = SmartThingsFlowHandler()
     flow.hass = hass
-    result = await flow.async_step_user({"access_token": str(uuid4())})
 
-    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "user"
-    assert result["errors"] == {"base": "base_url_not_https"}
+    result = await flow.async_step_user()
 
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "invalid_webhook_url"
+    assert result["description_placeholders"][
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5359319)